Havit.Blazor.Components.Web.Bootstrap.Documentation/Pages/Components/HxTreeViewDoc/HxTreeView_SimpleDemo.razor

```diff
@@ -0,0 +1,89 @@
+<div class="row">
+    <div class="col-4" style="height: 400px">
+        <HxTreeView
+            TValue="Directory"
+            OnItemSelected="@(p => SelectedDirectory = p)"
+            Items="@FileSystem"
+            ContentSelector="@(p => p.Name)"
+            ChildrenSelector="@(p => p.SubDirectories)"
+            IconSelector="@(p => p.Icon)"/>
+    </div>
+    <div class="col-4">
+        @if (SelectedDirectory == null)
+        {
+            <p>No dirrectory selected</p>
+        }
+        else
+        {
+            <p>Dirrectory <b>@SelectedDirectory.Name</b> selected.</p>
+        }
+    </div>
+</div>
+
+@code {
+
+    public Directory SelectedDirectory { get; set; }
+
+    public Directory[] FileSystem { get; set; }
+
+    protected override void OnInitialized()
+    {
+        FileSystem = new[]
+        {
+            new Directory("C:\\", nameof(BootstrapIcon.Folder))
+            {
+                SubDirectories = new[]
+                {
+                    new Directory("Program Files", nameof(BootstrapIcon.Gem))
+                    {
+                        SubDirectories = new[] { new Directory("Visual Studio 2022"), new Directory("Rider 2021.3"), new Directory("VS Code"), new Directory("Borland Pascal") }
+                    },
+                    new Directory("Users", nameof(BootstrapIcon.Person))
+                    {
+                        SubDirectories = new[] { new Directory("Vladimir"), new Directory("Jhosav"), new Directory("Ben"), new Directory("Admin") }
+                    },
+                    new Directory("Data", nameof(BootstrapIcon.Person))
+                    {
+                        SubDirectories = new[] { new Directory("Open Office"), new Directory("Git", nameof(BootstrapIcon.Magic)), new Directory("Autocad") }
+                    },
+                }
+            },
+            new Directory("D:\\", BootstrapIcon.Folder2.Name)
+            {
+                SubDirectories = new[]
+                {
+                    new Directory("Games", nameof(BootstrapIcon.Play))
+                    {
+                        SubDirectories = new[] { new Directory("Cyberpunk 2077"), new Directory("The Witcher 3"), new Directory("Tetris"), new Directory("Little Big Adventure 2: Twinsen's Odyssey") }
+                    },
+                    new Directory("Music", nameof(BootstrapIcon.FileMusic))
+                    {
+                        SubDirectories = new[] { new Directory("Queen"), new Directory("Metallica"), new Directory("AC/DC") }
+                    }
+                }
+            }
```

Comment:
@sshulik The way you are creating the icons supports my idea of having the `IconSelector` return `IconBase` directly.

Reply:
refactored